### PR TITLE
[APP-721] Render widget and global settings

### PR DESCRIPTION
**Pull request by nirbhayel**

### CODE REVIEW

**Review by pkniazevych — CHANGES_REQUESTED** [review]
Please, check the comment.

File: modules/widget/module.php [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+		wp_localize_script(
+			'ea11y-widget',
+			'ea11yWidgetData',
+			[
+				'iconSettings' => wp_json_encode( get_option( 'ea11y_widget_icon_settings' ) ),
+				'menuSettings' => wp_json_encode( get_option( 'ea11y_widget_menu_settings' ) ),
+				'wpRestNonce' => wp_create_nonce( 'wp_rest' )
+			]
+		);
```
[the remark this answers is not in the document]
Comment (nirbhayel) [reply modules/widget/module.php]: Updated @pkniazevych

File: classes/services/client.php [hunk 2]
```diff
@@ -56,9 +56,19 @@ public static function get_site_info(): array {
...
+			'webhook_url' => self::webhook_endpoint(),
 		];
 	}
 
+	/**
+	 * Log update endpoint
+	 * @return string
+	 */
+	private static function webhook_endpoint(): string {
+		$blog_id            = get_current_blog_id();
```
Comment (bainternet) [inline classes/services/client.php:68]:
```suggestion
		$blog_id = get_current_blog_id();
```

File: classes/services/client.php [hunk 3]
```diff
@@ -147,13 +157,20 @@ protected function request( $method, $endpoint, $args = [] ) {
...
 
-		$body = json_decode( $body );
+		/**
+		 * product/widget endpoint returns javascript hence we
+		 * escape decoding for the data received in body from
+		 * this endpoint.
+		 */
+		if ( ! strpos( $endpoint, 'widget' ) ) {
+			$body = json_decode( $body );
+		}
```
Comment (bainternet) [inline classes/services/client.php:173]: why do we need this? we will never call the widget from the service client directly

File: modules/widget/module.php [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+	private function get_widget_url() : string {
+		return apply_filters( 'ea11y_widget_url', '' ); // TODO: add public url
+	}
+
+	/**
+	 * Module constructor.
+	 */
+	public function __construct() {
+		add_action( 'wp_enqueue_scripts', [ $this, 'enqueue_accessibility_widget' ] );
+	}
```
Comment (bainternet) [inline modules/widget/module.php:59]: this should be enqueue only if connected

**Review by bainternet — CHANGES_REQUESTED** [review]
left some comments

File: modules/widget/module.php (hunk 4 above)
Comment (nirbhayel) [reply modules/widget/module.php:59]: @bainternet Should there be a check for plan data as well? If the pan data is not available, then the widget should not load.

Comment (bainternet) [reply modules/widget/module.php:59]: Yes also the pr is not fixed in both comments

File: modules/widget/module.php [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+	 * @return void
+	 * @throws Exception
+	 */
+	public function enqueue_accessibility_widget () : void {
+
+		if ( ! Connect::is_connected() ) {
+			return;
+		}
+
+		if ( empty ( Settings::PLAN_DATA ) && ! isset( $plan_data->public_api_key ) ) {
```
Comment (bainternet) [inline modules/widget/module.php:32]: this must be wrong ```empty ( Settings::PLAN_DATA )```

File: modules/widget/module.php [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+				return;
+		}
+
+		$plan_data = Settings::get( Settings::PLAN_DATA );
```
Comment (bainternet) [inline modules/widget/module.php:36]: and this should probably be above it

**Review by bainternet — CHANGES_REQUESTED** [review]
see comments

**Review by bainternet — APPROVED** [review]

**Pull request merged**